pkg/controller/master/backup/backup_target.go

```diff
@@ -79,32 +80,79 @@ func (h *TargetHandler) OnBackupTargetChange(key string, setting *harvesterv1.Se
 		return setting, err
 	}
 
-	// Since we remove S3 access key id and secret access key after S3 backup target has been verified,
-	// we stop the controller to reconcile it again.
-	if target.Type == settings.S3BackupType && (target.SecretAccessKey == "" || target.AccessKeyID == "") {
-		return nil, nil
-	}
+	logrus.Debugf("backup target change:%s:%s", target.Type, target.Endpoint)
 
-	if err = h.updateLonghornTarget(target); err != nil {
-		return nil, err
-	}
+	switch target.Type {
+	case settings.S3BackupType:
+		// Since S3 access key id and secret access key are stripped after S3 backup target has been verified
+		// in reUpdateBackupTargetSettingSecret
+		// stop the controller to reconcile it again.
+		if target.SecretAccessKey == "" && target.AccessKeyID == "" {
+			return nil, nil
+		}
+
+		if err = h.updateLonghornTarget(target); err != nil {
+			return nil, err
+		}
 
-	if target.Type == settings.S3BackupType {
 		if err = h.updateBackupTargetSecret(target); err != nil {
 			return nil, err
 		}
-	}
 
-	return h.updateBackupTargetSetting(setting.DeepCopy(), target, err)
+		return h.reUpdateBackupTargetSettingSecret(setting, target, err)
```

Comment:
I think we can just update the secrets on the validation webhook and strip the credentials through mutation webhook, and therefore the controller doesn't need to reconcile it.

Reply:
This idea may be verified and committed in next PR.
The secrets are used to create secrets in api server first, finally be stripped, not sure if mutation webhook works.

Reply:
keep now, wait for further discuss.